pxf/pxf-jdbc/src/main/java/org/apache/hawq/pxf/plugins/jdbc/JdbcPartitionFragmenter.java

```diff
@@ -0,0 +1,284 @@
+    }
+
+    /**
+     * Constructor for JdbcPartitionFragmenter.
+     *
+     * @param inConf input data such as which Jdbc table to scan
+     * @throws JdbcFragmentException
+     */
+    public JdbcPartitionFragmenter(InputData inConf) throws JdbcFragmentException {
+        super(inConf);
+        if(inConf.getUserProperty("PARTITION_BY") == null )
+            return;
+        partition_by = inConf.getUserProperty("PARTITION_BY").split(":");
+        partitionColumn = partition_by[0];
+        partitionType = PartitionType.getType(partition_by[1]);
+
+        range = inConf.getUserProperty("RANGE").split(":");
+
+        //parse and validate parameter-INTERVAL
+        if (inConf.getUserProperty("INTERVAL") != null) {
+            interval = inConf.getUserProperty("INTERVAL").split(":");
+            intervalNum = Integer.parseInt(interval[0]);
+            if (interval.length > 1)
+                intervalType = IntervalType.type(interval[1]);
+        }
+        if (intervalNum < 1)
+            throw new JdbcFragmentException("The parameter{INTERVAL} must > 1, but actual is '" + intervalNum+"'");
+    }
+    /**
+     * Returns statistics for Jdbc table. Currently it's not implemented.
+     */
+    @Override
+    public FragmentsStats getFragmentsStats() throws Exception {
+        throw new UnsupportedOperationException("ANALYZE for Jdbc plugin is not supported");
+    }
+
+
+
+    /**
+     * Returns list of fragments containing all of the
+     * Jdbc table data.
+     *
+     * @return a list of fragments
+     */
+    @Override
+    public List<Fragment> getFragments() throws Exception {
+        switch (partitionType) {
+            case DATE: {
+                SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd");
+                Date t_start = df.parse(range[0]);
+                Date t_end = df.parse(range[1]);
+                int curr_interval = intervalNum;
+
+                Calendar frag_start = Calendar.getInstance();
+                Calendar c_end = Calendar.getInstance();
+                frag_start.setTime(t_start);
+                c_end.setTime(t_end);
+                while (frag_start.before(c_end)) {//|| frag_start.compareTo(c_end) == 0) {
+                    Calendar frag_end = (Calendar) frag_start.clone();
+                    switch (intervalType) {
+                        case DAY:
+                            frag_end.add(Calendar.DAY_OF_MONTH, curr_interval);
+                            break;
+                        case MONTH:
+                            frag_end.add(Calendar.MONTH, curr_interval);
+                            break;
+                        //case YEAR:
+                        default:
+                            frag_end.add(Calendar.YEAR, curr_interval);
+                            break;
+                    }
+                    if (frag_end.after(c_end)) frag_end = (Calendar) c_end.clone();

```

Comment:
for readability, please break this statement to two lines.
```
  if ... {
    ...
  }
```
https://cwiki.apache.org/confluence/display/HAWQ/PXF+Coding+Conventions#PXFCodingConventions-OneLinersinBrackets